features/share/impl/src/main/kotlin/io/element/android/features/share/impl/ShareIntentHandler.kt

```diff
@@ -58,16 +58,16 @@ class DefaultShareIntentHandler(
         onPlainText: suspend (String) -> Boolean,
     ): Boolean {
         val type = intent.resolveType(context) ?: return false
+        val uris = getIncomingUris(intent, type)
         return when {
-            type == MimeTypes.PlainText -> handlePlainText(intent, onPlainText)
+            uris.isEmpty() && type == MimeTypes.PlainText -> handlePlainText(intent, onPlainText)
```

Comment:
Thanks. Now I am wondering why we are doing so much filtering on the types. This is maybe not necessary?
the code could be simplified by
```kotlin
        return if (uris.isEmpty() && type == MimeTypes.PlainText) {
            handlePlainText(intent, onPlainText)
        } else {
            val result = onUris(uris)
            revokeUriPermissions(uris.map { it.uri })
            result
        }
```
Not a blocker and not related to the issue that this PR is fixing, I am just thinking out loud.

Reply:
I'm not sure either, I just kept it like that because I couldn't remember the reasoning 😅 .